generic_worker.go

```diff
@@ -35,9 +37,13 @@ func (g *Generic) processNextWorkItem() bool {
 	defer g.queue.done(key)
 
 	holder := g.Controllers[key.gvk]
-	logger := g.logger.With(logz.NamespaceName(key.Namespace), holder.ZapNameField(key.Name))
+	logger := g.logger.With(logz.NamespaceName(key.Namespace),
+		holder.ZapNameField(key.Name), zap.Uint64(fieldIteration, g.iter))
+	defer func() {
+		g.iter++
```

Comment:
This must use atomic instructions. `sync/atomic.AddUint64(&g.iter, 1)` will return the new value. Just use it in the logger construction?